dss/util/__init__.py

```diff
@@ -134,3 +135,53 @@ def get_gcp_credentials_file():
     tf.write(resp['SecretString'])
     tf.flush()
     return tf
+
+
+def multipart_parallel_upload(
+        s3_client: typing.Any,
+        bucket: str,
+        key: str,
+        src_file_handle: typing.BinaryIO,
+        *,
+        size: int,
+        part_size: int,
+        content_type: str=None,
+        metadata: dict=None,
+        parallelization_factor=8) -> typing.Sequence[dict]:
```

Comment:
Should this default to a multiple of the system cpu count?

Reply:
Reasonable, yes, see other comment :)